tmt/utils.py

```diff
@@ -2026,6 +2026,113 @@ def environment_files_to_dict(
     return result
 
 
+def environment_from_spec(
+        *,
+        raw_fmf_environment: Any = None,
+        raw_fmf_environment_files: Any = None,
+        raw_cli_environment: Any = None,
+        raw_cli_environment_files: Any = None,
+        file_root: Optional[Path] = None,
+        key_address: Optional[str] = None,
+        logger: tmt.log.Logger) -> EnvironmentType:
+    """
+    Extract environment variables from various sources.
+
+    Combines various raw sources into a set of environment variables. Calls
+    necessary functions to process environment files, dictionaries and CLI
+    inputs.
+
+    All inputs are optional, and there is a clear order of preference, which is,
+    from the most prefered:
+
+    * ``--environment`` CLI option (``raw_cli_environment``)
+    * ``--environment-file`` CLI option (``raw_cli_environment_files``)
+    * ``environment`` fmf key (``raw_fmf_environment``)
+    * ``environment-file`` fmf key (``raw_fmf_environment_files``)
+
+    :param raw_fmf_environment: content of ``environment`` fmf key. ``None``
+        and a dictionary are accepted.
+    :param raw_fmf_environment_files: content of ``environment-file`` fmf key.
+        ``None`` and a list of paths are accepted.
+    :param raw_cli_environment: content of ``--environment`` CLI option.
+        ``None``, a tuple or a list are accepted.
+    :param raw_cli_environment_files: content of `--environment-file`` CLI
+        option. ``None``, a tuple or a list are accepted.
+    :raises NormalizationError: when an input is of a type which is not allowed
+        for that particular source.
+    """
+
+    key_address_prefix = f'{key_address}:' if key_address else ''
+
+    from_fmf_files: EnvironmentType = {}
+    from_fmf_dict: EnvironmentType = {}
+    from_cli_files: EnvironmentType = {}
+    from_cli: EnvironmentType = {}
+
+    if raw_fmf_environment_files is None:
+        pass
+
+    elif isinstance(raw_fmf_environment_files, list):
+        raw_fmf_environment_files = raw_fmf_environment_files or []
```

Comment:
Why is this necessary? Don't we already have a list instance?

Reply:
Correct, no longer needed, removed in b4776162